### PR TITLE
101522656 create security groups

### PR DESCRIPTION
**Pull request by s-bortolussi**
Add an interface and Spring implementation for the Create Security Group endpoint (POST /v2/security_groups).
[#101522656]

### CODE REVIEW

File: cloudfoundry-client/src/main/java/org/cloudfoundry/client/v2/securitygroups/_RuleEntity.java [hunk 1]
```diff
@@ -47,4 +47,25 @@
     @JsonProperty("protocol")
     abstract String getProtocol();
 
+    /**
+     * The control signal for icmp
+     */
+    @JsonProperty("type")
+    @Nullable
+    abstract Byte getType();
```
Comment (nebhale) [inline cloudfoundry-client/src/main/java/org/cloudfoundry/client/v2/securitygroups/_RuleEntity.java]: @s-bortolussi Should be an `Integer`.  I realize that this isn't accurate with regards to ICMP, but it is accurate with respects to the expected JSON payload.

File: cloudfoundry-client/src/main/java/org/cloudfoundry/client/v2/securitygroups/_RuleEntity.java [hunk 2]
```diff
@@ -47,4 +47,25 @@
...
+
+    /**
+     * The control signal for icmp
+     */
+    @JsonProperty("code")
+    @Nullable
+    abstract Byte getCode();
```
Comment (nebhale) [inline cloudfoundry-client/src/main/java/org/cloudfoundry/client/v2/securitygroups/_RuleEntity.java]: @s-bortolussi Should be an `Integer`.  I realize that this isn't accurate with regards to ICMP, but it is accurate with respects to the expected JSON payload.  I'll fix.

File: cloudfoundry-client/src/main/java/org/cloudfoundry/client/v2/securitygroups/_CreateSecurityGroupRequest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    abstract List<RuleEntity> getRules();
+
+    /**
+     * The list of associated spaces
+     */
+    @JsonProperty("space_guids")
+    @Nullable
+    abstract List<String> getSpaceIds();
+
+
```
Comment (nebhale) [inline cloudfoundry-client/src/main/java/org/cloudfoundry/client/v2/securitygroups/_CreateSecurityGroupRequest.java]: @s-bortolussi Watch your formatting here; too many empty lines.

**Comment by nebhale** [comment]
@s-bortolussi Overall a _really_ good contribution.  One thing to keep an eye on if you plan on continuing to contribute is commits and messages.  For a given "logical change" (e.g. Pivotal Tracker issue) we expect a single commit.  You can develop any way you choose, but please squash multiple commits into a single commit before submission.  In addition, we're pretty strict on how commit messages should be structured.  We follow the [generally accepted structure](http://chris.beams.io/posts/git-commit/) for Git commit messages, but when in doubt, take a look at [some](https://github.com/cloudfoundry/cf-java-client/commit/4479d8e6f60c98e5df7836964bd4a5009606e8b6) [other](https://github.com/cloudfoundry/cf-java-client/commit/e308e8178130106d66b6eb675684df63d37e4889) [commits](https://github.com/cloudfoundry/cf-java-client/commit/8445cbdf57e0700b3f4c7faee6d913ebdb61886f). 

As I said, a really good contribution and I hope to see more from you if you're game.  Thanks for the help!

**Comment by s-bortolussi** [comment]
@nebhale Thanks for your changes and feedback.

**Pull request closed without merging**